src/app/_components/ListboxOptions.tsx

```diff
@@ -1,22 +1,41 @@
-import { ListboxOptions as HeadlessUIListboxOptions } from '@headlessui/react'
+import type { ElementType } from 'react'
+
+import {
+  ListboxOptions as HeadlessUIListboxOptions,
+  type ListboxOptionsProps as HeadlessUIListboxOptionsProps,
+} from '@headlessui/react'
 import { clsx } from 'clsx'
 
+type HeadlessUIAnchorProps = NonNullable<
+  HeadlessUIListboxOptionsProps['anchor']
+>
+type Position = Extract<HeadlessUIAnchorProps, string>
+
 type ListboxOptionsProps = {
-  position?: 'left' | 'right'
+  as?: ElementType
+  position?: Position
+  gap?: number
+  matchButtonWidth?: boolean
   children: React.ReactNode
 }
 
+const DISTANCE_FROM_TRIGGER_IN_PX = 8
```

Comment:
Since we have `matchButtonWidth` - Shall we rename this to `DISTANCE_FROM_BUTTON_IN_PX`?
```suggestion
const DISTANCE_FROM_BUTTON_IN_PX = 8
```

Reply:
I like that!